### PR TITLE
PA-606 Add base typescript eslint and tsconfig

### PR DESCRIPTION
**Pull request by martiantim**
Base typescript eslint. Adds "noImplicitOverride" in addition to existing rules

### CODE REVIEW

File: eslint-ts.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    "extends": "maven",
+    "env": {
+        "node": true
+    },
+    "parser": "@typescript-eslint/parser",
+    "plugins": ["@typescript-eslint"],
```
Comment (stanimoto) [inline eslint-ts.js:7]: What do you think about moving the dependency to these packages into this repo, assuming there is no negative impact on other node projects?

**Review by stanimoto — COMMENTED** [review]
Quick question about the depedencies, but it looks good to me.

File: base-tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "compilerOptions": {
+        "target": "es2020",
```
Comment (yonran) [inline base-tsconfig.json:3]: [By default](https://github.com/microsoft/TypeScript/blob/v4.4.3/src/compiler/utilitiesPublic.ts#L19-L20), if [lib](https://www.typescriptlang.org/tsconfig/#lib) is omitted, then the lib will be set to [lib.es2020.full.d.ts](https://github.com/microsoft/TypeScript/blob/main/lib/lib.es2020.full.d.ts), which includes dom that we probably don’t want in node apps (unless we’re using jsdom). By contrast, [@tsconfig/node14/tsconfig.json](https://github.com/tsconfig/bases/blob/main/bases/node14.json) sets `"lib": ["es2020"]`, and [@tsconfig/node16/tsconfig.json](https://github.com/tsconfig/bases/blob/main/bases/node16.json) sets `"lib": ["es2021"]`

Maybe we should set `"lib": ["es2020"]` with a comment that this is for node14.

File: eslint-ts.js (hunk 1 above)
Comment (martiantim) [reply eslint-ts.js:7]: Good idea, done

File: base-tsconfig.json (hunk 2 above)
Comment (martiantim) [reply base-tsconfig.json:3]: Makes sense, done

**Review by stanimoto — APPROVED** [review]

**Pull request merged**